### PR TITLE
Transition to Python3

### LINKED ISSUE #5: langkit is python 2.7 only?
Attempting to build with python 3.5+:
```
Traceback (most recent call last):
  File "ada/manage.py", line 14, in <module>
    from langkit.libmanage import ManageScript
  File "/construction/libadalang/libadalang-611f857/ada/../langkit/langkit/libmanage.py", line 17, in <module>
    from langkit.compile_context import Verbosity
  File "/construction/libadalang/libadalang-611f857/ada/../langkit/langkit/compile_context.py", line 1358
    key=lambda (s, n): n)
               ^
SyntaxError: invalid syntax
```

Python 2.7 is nearly EOL.  In my opinion, no new development should target python 2.7.
At the very least, if this really is a restriction, it should be boldly proclaimed on the README as that is very important build information.

### PR DESCRIPTION
**Pull request by pmderodat**
Fixes #5

### CODE REVIEW

File: langkit/expressions/base.py [hunk 1]
```diff
@@ -1705,7 +1707,27 @@ def render_public_ada_constant(self):
         return self.ada_value
 
     def render_python_constant(self):
-        return repr(self.value)
+        # Stick to ASCII in generated sources
```
[the remark this answers is not in the document]
Comment (pmderodat) [reply langkit/expressions/base.py:1710]: When Langkit runs on Python3, `repr('é')` returns `'é'`, and thus Python2 interpreters complain on the generated Python bindings, as they contain non-ASCII sources. I’ll extend the comment with this rationale, thanks.

[the remark this answers is not in the document]
Comment (raph-amiard) [reply langkit/expressions/base.py:1710]: Ok! :+1:

File: README.md [hunk 2]
```diff
@@ -24,8 +24,8 @@ To use Langkit:
 - The mako template system for Python (see `REQUIREMENTS.dev`)
 - Clang-format
 
-Note that for now, Langkit requires a Python 2.7 interpreter. That being said,
-Python3 support is planned and we should not be very far from it.
+Note that for now, Langkit requires a Python 3.7+ interpreter. Python2 is no
```
[the remark this answers is not in the document]
Comment (pmderodat) [reply README.md]: Fixed, thanks.

File: langkit/expressions/base.py [hunk 3]
```diff
@@ -480,6 +476,12 @@ def diagnostic_context(self):
     def __init__(self):
         self.location = extract_library_location()
 
+    def __hash__(self):
+        # AbstractExpression instances appear heavily as memoized functions
+        # arguments, so we need them to be hashable. There is no need for
+        # deterministic hashing in this context.
```
[the remark this answers is not in the document]
Comment (pmderodat) [reply langkit/expressions/base.py]: Good point, fixed.

File: langkit/expressions/structs.py [hunk 4]
```diff
@@ -255,9 +255,14 @@ def field_or_lookup(field):
             )
 
         def _iter_ordered(self):
+            def key(field_expr):
+                field, _ = field_expr
+                return field.name
+
             return sorted(
                 [(field, expr) for field, expr in self.assocs.items()],
-                key=lambda (field, _): field.name)
```
[the remark this answers is not in the document]
Comment (pmderodat) [reply langkit/expressions/structs.py:260]: Applied to this and other instances.

**Pull request merged**